android/lib/src/main/java/org/vosk/android/SpeechService.java

```diff
@@ -227,31 +253,91 @@ public void run() {
                 if (nread < 0)
                     throw new RuntimeException("error reading audio buffer");
 
-                if (recognizer.acceptWaveForm(buffer, nread)) {
-                    final String result = recognizer.getResult();
-                    mainHandler.post(() -> listener.onResult(result));
-                } else {
-                    final String partialResult = recognizer.getPartialResult();
-                    mainHandler.post(() -> listener.onPartialResult(partialResult));
+                // Adds the data read from the AudioRecorder to the queue of chunks of audio
+                try {
+                    transcriptionThread.recordingChunksQueue.put(new RecordingChunk(buffer, nread));
```

Comment:
Just checking that we are doing this on purpose
* queue.offer() is a synchronous attempt at add the element, which fails if the queue is full
* queue.put() is a synchronous attempt to add the element which will block until the queue is no longer full

I think the better idea would be to:
* Check the queue remaining capacity
* Then use the put to have a blocking insert
* If it has a couple of elements lagging output that we are lagging (a method onTranscriptionDelayed() or something like that)
* If it is full output that we are losing data (onDataProcessingFailed() or something like that)